### ISSUE TITLE
IMPORTANT - Add attribute to person so that their last travel time is known (date that travel was initiated)

### ISSUE THREAD

**Issue by maciekboni** [bonilab/Temple-Malaria-Simulation#2]
We would like each person to have two additional attributes:

int day_that_last_trip_was_initiated;
int day_that_last_trip_outside_district_was_initiated;

Every time someone travels (initiates a new trip), we update these two fields with the day that this occurs.

If a person's trip is outside their district/region/province (i.e. whatever admin-1 or admin-2 unit is currently being used), then we update the second attribute above as well with the current date.

At the end of the simulation, we loop through all people in all districts and report, per district, what percentage of people travelled in the last 30 days, the last 60 days, and the last 90 days.   Output as a CSV is probably best.  We should also output the district population size into this CSV.

**Comment by merlinvn** [comment]
This fearture has been implemented in 0292fa1504a651f3b77f3c39fcb64bab0f374e92

## Note:
Add TravelTracking feature to keep track and report the percentage of population travels in the past 30, 60, or 90 days.

- Two new attributes to the `Person` class to track the initiation days of the last trip and the last trip outside the district: `day_that_last_trip_was_initiated` and `day_that_last_trip_outside_district_was_initiated`.
- Compile-time flag `ENABLE_TRAVEL_TRACKING` to toggle the new travel tracking feature, allowing it to be enabled or disabled at compile time for performance considerations.
- Added TravelTrackingReporter to report (by district) the percentages of the population that have traveled in the past 30, 60, or 90 days to csv file named `travel_tracking_{job_id}.csv`
- Noted that, if a person traveled in the past 30 days, he will also be counted as traveled in the past 60 days and 90 days.

### Usage

To turn on the `TravelTracking` feature, you need to specify the `-DENABLE_TRAVEL_TRACKING=ON` option to `cmake` when building the simulation:

```sh
# Generate the build folder with cmake and turn on the `TravelTracking` feature
make generate ENABLE_TRAVEL_TRACKING=ON
make build
```

**Comment by maciekboni** [comment]
Thank you Nguyen.  We can test this together next week.
